# University of Niagara Falls  
## Master of Science in Data Analytics  

---

# **CREDIT DEFAULT RISK ANALYSIS REPORT**  
### A Machine Learning Classification Study

---

### **Course Name:**  
CPSC‑610: Python for Data Analysis  

### **Instructor:**  
Dr. Mohannad Al Mousa  

### **Date of Submission:**  
June 14, 2026  

### **Student Team:**  
1. Manpreet Kaur  
2. Nicolas Ricardo Ysa Pelaez  
3. Daniela Montilla Fernandez  
4. Jose Alberto Martinez Morales  
5. Sravani Maguluru  

---

# 1. Project Overview

This project examines credit default risk using two complementary datasets: a loan applications dataset used for predictive modeling, and a credit inquiries dataset used for business insights. Together, these datasets provide a comprehensive view of borrower behavior, credit demand, and repayment outcomes.

---

## Applications Dataset. Used for Predictive Modeling

Each row in the applications dataset represents a completed loan application with detailed borrower, loan, and credit profile information.  
The dataset includes the following key fields:

- application_id — Unique identifier for each loan application  
- requested_amount_usd — Loan amount requested  
- repayment_term_text — Repayment duration as text  
- interest_rate_text — Interest rate as text  
- monthly_installment_usd — Scheduled monthly payment  
- last_payment_amount_usd — Most recent payment amount  
- risk_grade_band and risk_grade_detail — Credit risk categories  
- employment_tenure_text — Borrower employment length  
- housing_status — Housing situation  
- annual_income_usd — Reported annual income  
- income_verification_status — Verification category  
- origination_month — Month the loan originated  
- outcome_status — Final loan outcome  
- loan_purpose — Borrower-stated purpose  
- region_code — Geographic region  
- debt_to_income_ratio — Borrower DTI  
- credit_score_low and credit_score_high — Credit score range  
- open_credit_lines — Number of open accounts  
- public_records_count — Number of derogatory public records  
- revolving_utilization_text — Revolving credit utilization  

Because this dataset contains the final loan outcome, it is used to build the predictive model.

---

## Prediction Target

The goal of the predictive component is to determine whether a loan application results in default.  
The target variable is defined as:

- target_default = 1 if outcome_status == "Charged Off"  
- target_default = 0 otherwise  

This converts the problem into a binary classification task suitable for machine learning.

---

## Inquiries Dataset. Used for Business Insights

The inquiries dataset contains information about credit inquiries submitted by potential borrowers.  
It includes the following fields:

- requested_amount_usd — Amount requested during inquiry  
- application_date — Date of inquiry  
- request_title — Free-text description of the request  
- screening_score — Preliminary risk score  
- debt_to_income_text — DTI expressed as text  
- region_code — Geographic region  
- employment_tenure_text — Employment length  
- policy_code — Policy indicator  

This dataset does not contain loan outcomes and therefore cannot be used for supervised learning.  
Instead, it supports business insights by revealing patterns in borrower demand, screening score distributions, regional inquiry trends, and employment characteristics at inquiry time.

---

## Model Evaluation Approach


The model will be evaluated using three complementary methods:

1. **Confusion Matrix**  
   Used to understand the distribution of correct and incorrect predictions.

2. **A false-positive–focused metric**  
   This directly supports the business objective of minimizing cases where the model incorrectly predicts that a borrower will repay when they actually default.

3. **ROC–AUC Curve**  
   Used to assess overall model discrimination and to help analyze potential overfitting or underfitting.

---

## Project Workflow

The analysis follows a complete data science pipeline:

### 1. Data Cleaning  
Both datasets are cleaned thoroughly by:  
- Handling missing values  
- Removing duplicates  
- Converting text fields to numeric  
- Standardizing categorical values  
- Detecting and treating outliers  
- Removing leakage variables  
- Ensuring correct data types  

### 2. Exploratory Data Analysis (EDA)  
EDA is performed separately on each dataset to uncover:  
- Distribution of key variables  
- Relationships between borrower characteristics and default risk  
- Regional and behavioral patterns in inquiry activity  
- Correlations among financial indicators  

### 3. Feature Engineering  
The applications dataset is transformed into a modeling-ready format through:  
- Creation of numeric features (for example, employment tenure, repayment term)  
- Cleaning and scaling of financial variables  
- Encoding of categorical variables  
- Removal of leakage and post-outcome fields  

### 4. Model Building  
At least two classification models are developed:  
- Logistic Regression (baseline)  
- Random Forest (non-linear model)  

Models are trained using a stratified train/test split and a preprocessing pipeline.

### 5. Model Evaluation  
Models are evaluated using:  
- Confusion matrix  
- A false-positive–focused metric aligned with the business objective  
- ROC–AUC curve to assess model separation and diagnose overfitting or underfitting
  
### 6. Business Insights  
Insights are drawn from both datasets to answer questions such as:  
- Which borrower characteristics are most associated with default?  
- Which loan purposes or regions exhibit higher risk?  
- What patterns emerge in inquiry behavior?  
- How can lenders adjust policies or thresholds to reduce losses?  

### 7. Conclusion and Recommendations  
The project concludes with:  
- A summary of findings  
- The recommended model  
- Actionable business recommendations  
- Limitations and opportunities for future work  

---

## Purpose and Relevance

Credit default prediction is essential for effective risk management.  
Accurate models help lenders:  
- Reduce financial losses  
- Improve approval decisions  
- Identify high-risk borrower segments  
- Support pricing and policy strategies  

By combining predictive modeling with inquiry-level insights, this project provides a comprehensive framework for understanding and managing credit risk.

# 2. Data Cleaning

This section prepares both datasets for analysis by identifying and resolving issues related to missing values, duplicates, incorrect data types, inconsistent formatting, and outliers.  
The cleaning process is divided into two parts:

- **2A. Applications Dataset (Prediction Dataset)**  
- **2B. Inquiries Dataset (Business Insights Dataset)**  

Each subsection includes diagnostics, visualizations, and documented decisions.

---

# 2A. Data Cleaning — Applications Dataset (Prediction Dataset)

The applications dataset contains borrower, loan, and credit profile information.  
Because this dataset is used for predictive modeling, it must be cleaned thoroughly to avoid data leakage, incorrect model behavior, or biased results.

---

## 2A.1 Initial Structure and Diagnostics

We begin by inspecting the dataset structure, data types, missing values, and duplicates.


In [15]:
import pandas as pd
import numpy as np
apps = pd.read_csv('../data/creditrisk_applications_final.csv')
apps.info()

<class 'pandas.DataFrame'>
RangeIndex: 85000 entries, 0 to 84999
Data columns (total 22 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   application_id              85000 non-null  int64  
 1   requested_amount_usd        85000 non-null  float64
 2   repayment_term_text         85000 non-null  str    
 3   interest_rate_text          85000 non-null  float64
 4   monthly_installment_usd     85000 non-null  float64
 5   last_payment_amount_usd     85000 non-null  float64
 6   risk_grade_band             85000 non-null  str    
 7   risk_grade_detail           85000 non-null  str    
 8   employment_tenure_text      76324 non-null  str    
 9   housing_status              85000 non-null  str    
 10  annual_income_usd           84320 non-null  float64
 11  income_verification_status  85000 non-null  str    
 12  origination_month           85000 non-null  str    
 13  outcome_status              85000 non-null

In [16]:
# Checking null values 
apps.isna().sum().sort_values(ascending=False)

employment_tenure_text        8676
revolving_utilization_text     892
debt_to_income_ratio           709
annual_income_usd              680
application_id                   0
requested_amount_usd             0
last_payment_amount_usd          0
monthly_installment_usd          0
interest_rate_text               0
repayment_term_text              0
housing_status                   0
income_verification_status       0
risk_grade_detail                0
risk_grade_band                  0
outcome_status                   0
origination_month                0
region_code                      0
loan_purpose                     0
credit_score_low                 0
credit_score_high                0
open_credit_lines                0
public_records_count             0
dtype: int64

In [17]:
# Checking duplicates
apps.duplicated('application_id').sum()

np.int64(0)

In [18]:
apps.shape

(85000, 22)

### **Insight  Data Quality and Missingness Patterns**

The applications dataset is generally well‑structured, with **85,000 rows** and **22 columns**, and no duplicate `application_id` values.  
However, several fields contain meaningful missingness that must be addressed before modeling:

- **employment_tenure_text** has the highest missingness (**8,676 records**).  
  This suggests inconsistent reporting of employment history, which may reflect borrower instability or data collection gaps.

- **revolving_utilization_text**, **debt_to_income_ratio**, and **annual_income_usd** each have **600–900 missing values**.  
  These are financially important variables, so missingness must be handled carefully to avoid bias.

- All other fields are fully populated, including key credit indicators such as  
  `credit_score_low`, `credit_score_high`, `open_credit_lines`, and `public_records_count`.

- No duplicate applications were found, confirming dataset integrity.

**Interpretation:**  
The missingness is concentrated in employment and financial capacity variables, which are often correlated with borrower risk.  
This pattern suggests that borrowers with incomplete financial information may represent a distinct risk segment, and imputation strategies must be chosen carefully to avoid introducing bias.

## 2A.2 Duplicate Removal

In [19]:
# Checking for duplicates
apps = apps.drop_duplicates(subset='application_id')

### **Insight Duplicate Check on `application_id`**

The result shows:

- **0 duplicate rows**
- **All 85,000 application IDs are unique**

**Interpretation:**  
The applications dataset has strong structural integrity.  
Since each `application_id` appears exactly once, we can confidently treat each row as a unique loan application without needing to remove or consolidate duplicate entries.  
This ensures that the modeling dataset is not biased by repeated records and that each borrower’s application contributes equally to the analysis.

## 2A.3 Numeric and Categorical Diagnostics Before Cleaning

Before converting any fields, we first examine the dataset by separating **numeric** and **categorical** columns.  
This allows us to verify whether each variable has the correct datatype and identify fields that require transformation.

---

### **2A.3.1 Summary Statistics — Numeric Columns**

In [20]:
# Describe in numerical variables 
apps.select_dtypes(include=['int64', 'float64']).describe()

,application_id,requested_amount_usd,interest_rate_text,monthly_installment_usd,last_payment_amount_usd,annual_income_usd,credit_score_low,credit_score_high,open_credit_lines,public_records_count,revolving_utilization_text
count,8.500000e+04,85000.000000,85000.000000,85000.000000,85000.000000,8.432000e+04,85000.000000,85000.000000,85000.000000,85000.000000,84108.000000
mean,6.633685e+07,14950.852647,13.600630,446.071413,3604.090989,7.434189e+04,692.034412,696.034482,11.920871,0.248718,52.342891
std,2.482399e+07,8735.147461,4.886372,258.523612,6222.901104,7.732271e+04,29.787475,29.787839,5.670095,0.656437,24.293508
min,3.476740e+05,1000.000000,5.310000,30.120000,0.000000,0.000000e+00,660.000000,664.000000,1.000000,0.000000,0.000000
25%,4.881616e+07,8000.000000,9.990000,259.955000,308.220000,4.500000e+04,670.000000,674.000000,8.000000,0.000000,34.200000
50%,6.059760e+07,13050.000000,13.180000,382.870000,598.895000,6.300000e+04,685.000000,689.000000,11.000000,0.000000,52.800000
75%,7.635120e+07,20000.000000,16.550000,587.690000,4130.377500,9.000000e+04,705.000000,709.000000,15.000000,0.000000,70.900000
max,1.411120e+08,40000.000000,30.990000,1587.230000,40551.270000,8.900060e+06,845.000000,850.000000,70.000000,49.000000,136.400000


### **Insight Numeric Variable Summary**

The summary statistics for all numeric columns reveal that the dataset is generally well‑structured, with reasonable ranges and no obvious data corruption in the core financial and credit variables.

Key observations:

- **Loan amounts and payments** (`requested_amount_usd`, `monthly_installment_usd`, `last_payment_amount_usd`) show realistic ranges with no negative values, confirming correct numeric formatting.

- **Credit score fields** (`credit_score_low`, `credit_score_high`) fall within expected credit score boundaries (typically 300–850), indicating clean and valid values.

- **Open credit lines and public records** contain non‑negative integers, as expected for count variables.

- **Interest rate** and **revolving utilization** are stored as numeric values but represent **percentages**, meaning they must be converted to decimals before modeling.

- **Annual income** shows a wide but reasonable distribution, with no extreme outliers that would indicate data corruption.

**Interpretation:**  
The numeric fields are structurally sound and correctly typed.  
The only required transformation is converting percentage‑based numeric fields (interest rate and revolving utilization) into decimal form.  
No numeric variable appears to be misclassified as categorical, and no categorical variable appears to be incorrectly stored as numeric.

### **2A.3.2 Summary Statistics — Categorical Columns**

In [21]:
apps.select_dtypes(include=['object', 'string']).describe()

,repayment_term_text,risk_grade_band,risk_grade_detail,employment_tenure_text,housing_status,income_verification_status,origination_month,outcome_status,loan_purpose,region_code,debt_to_income_ratio
count,85000,85000,85000,76324,85000,85000,85000,85000,85000,85000,84291
unique,2,7,35,11,7,3,23,2,24,100,4391
top,36 months,C,C4,10+ years,MORTGAGE,Source Verified,Jul-2015,Fully Paid,debt_consolidation,CA,bad
freq,59223,25111,5289,26275,38111,36936,6989,42500,49344,11601,336


### **Insight Categorical Variable Summary**

The categorical summary reveals that the dataset contains a diverse set of borrower, loan, and credit‑related categories.  
Most categorical fields appear structurally sound, but several require standardization or datatype correction.

Key observations:

- **repayment_term_text** contains only two categories (“36 months” and “60 months”), confirming it should be treated as a **categorical variable**, not numeric.  
  Minor formatting inconsistencies (capitalization, spacing) may require standardization.

- **employment_tenure_text** shows a wide range of employment duration labels (e.g., “10+ years”, “< 1 year”, “3 years”).  
  These represent **ordered categories**, not numeric values, and must be cleaned for consistency.

- **debt_to_income_ratio** is stored as **mixed text**, containing:  
  - numeric strings (“18”, “24”, “14.88”)  
  - corrupted values (“bad”)  
  - thousands of unique text entries  
  This confirms the column is incorrectly typed and must be cleaned and converted to numeric.

- **risk_grade_band**, **risk_grade_detail**, **housing_status**, **loan_purpose**, and **region_code** show clean, well‑defined category sets with no obvious anomalies.

- **origination_month** is stored as text but represents a **date‑like categorical variable**.  
  It may be useful to convert this into a proper datetime or extract month/quarter features during feature engineering.

**Interpretation:**  
The categorical fields are generally well‑structured, but several require cleaning or standardization.  
The most significant issue is the `debt_to_income_ratio` column, which must be converted from mixed text to numeric.  
Repayment term and employment tenure should remain categorical but need consistent formatting.


## 2A.4 Cleaning Based on Diagnostics for Numerical Variables

This section applies the cleaning rules derived from the numerical diagnostics and prepares the dataset for modeling. The following steps were performed:

---

### 1. Remove application_id (non-predictive identifier)

The variable application_id is a unique identifier with no predictive value.  
It is removed from the dataset:

In [22]:
# Removing the application id
apps.drop(columns=['application_id'], inplace=True)

### 2. Convert all _text numeric fields into proper numeric variables
Several numeric variables were originally stored as text and ended with the suffix _text.
For each of these variables:

Remove the _text suffix
Convert the cleaned column to numeric
Add the new numeric column to the dataset
Drop the original _text column

Variables cleaned:

interest_rate_text → interest_rate
revolving_utilization_text → revolving_utilization

Code used:

In [23]:
# Only numeric _text columns
numeric_text_cols = [
    'interest_rate_text',
    'revolving_utilization_text'
]

for col in numeric_text_cols:
    new_col = col.replace('_text', '')
    apps[new_col] = pd.to_numeric(apps[col], errors='coerce')
    apps.drop(columns=[col], inplace=True)

In [24]:
apps.select_dtypes(include=['int64', 'float64']).describe()

,requested_amount_usd,monthly_installment_usd,last_payment_amount_usd,annual_income_usd,credit_score_low,credit_score_high,open_credit_lines,public_records_count,interest_rate,revolving_utilization
count,85000.000000,85000.000000,85000.000000,8.432000e+04,85000.000000,85000.000000,85000.000000,85000.000000,85000.000000,84108.000000
mean,14950.852647,446.071413,3604.090989,7.434189e+04,692.034412,696.034482,11.920871,0.248718,13.600630,52.342891
std,8735.147461,258.523612,6222.901104,7.732271e+04,29.787475,29.787839,5.670095,0.656437,4.886372,24.293508
min,1000.000000,30.120000,0.000000,0.000000e+00,660.000000,664.000000,1.000000,0.000000,5.310000,0.000000
25%,8000.000000,259.955000,308.220000,4.500000e+04,670.000000,674.000000,8.000000,0.000000,9.990000,34.200000
50%,13050.000000,382.870000,598.895000,6.300000e+04,685.000000,689.000000,11.000000,0.000000,13.180000,52.800000
75%,20000.000000,587.690000,4130.377500,9.000000e+04,705.000000,709.000000,15.000000,0.000000,16.550000,70.900000
max,40000.000000,1587.230000,40551.270000,8.900060e+06,845.000000,850.000000,70.000000,49.000000,30.990000,136.400000


In [32]:
# Ensure full output (prevents truncation in Jupyter)
pd.set_option('display.max_rows', None)
pd.set_option('display.max_colwidth', None)

# Select ONLY string + categorical columns (future-proof)
cat_cols = apps.select_dtypes(include=['string', 'category']).columns

# Loop through each categorical variable and print value counts
for col in cat_cols:
    print(f"\n===== {col} =====")
    print(apps[col].value_counts(dropna=False))



===== repayment_term =====
repayment_term
36 months    59223
60 months    25777
Name: count, dtype: Int64

===== risk_grade_detail =====
risk_grade_detail
C4    5289
C1    5139
C3    5051
C2    5036
B5    4708
B4    4602
C5    4596
B3    4342
B1    3795
B2    3722
D1    3616
A5    3505
D2    3122
D3    2821
D4    2791
A4    2719
D5    2425
A1    2100
E1    1996
E2    1922
A3    1819
A2    1803
E3    1697
E4    1380
E5    1316
F1     836
F2     639
F3     513
F4     413
F5     340
G1     269
G2     215
G3     176
G4     160
G5     127
Name: count, dtype: int64

===== employment_tenure =====
employment_tenure
10+ years    26275
<NA>          8676
2 years       7260
< 1 year      6985
3 years       6647
1 year        5590
4 years       4964
5 years       4921
8 years       3912
7 years       3380
6 years       3257
9 years       3133
Name: count, dtype: Int64

===== housing_status =====
housing_status
mortgage    39118
rent        36217
own          9664
any             1
Name: count, dt

## 2A.5 Cleaning Based on Diagnostics for Categorical Variables

The categorical diagnostics revealed several structural inconsistencies, duplicated categories, formatting issues, and variables requiring datatype correction.  
This section documents the cleaning steps applied to ensure all categorical fields are standardized, interpretable, and ready for modeling.

---

### 1. Remove `_text` suffix from categorical variables

Some categorical variables were originally stored with a `_text` suffix.  
These were renamed to cleaner, modeling‑friendly versions:

- `repayment_term_text` → `repayment_term`  
- `employment_tenure_text` → `employment_tenure`

Only the name was changed — the underlying dtype remains as string/object.

---

### 2. Remove redundant categorical variable

`risk_grade_band` and `risk_grade_detail` encode the same information at different granularities.  
To avoid multicollinearity and redundancy:

- **`risk_grade_band` was removed**  
- **`risk_grade_detail` was retained** as the more detailed and informative variable

---

### 3. Standardize formatting across all categorical variables

Many categorical fields contained inconsistent capitalization, spacing, or duplicated categories.  
To ensure uniformity, the following transformations were applied:

- Trim leading/trailing spaces  
- Convert all text to lowercase  
- Normalize inconsistent variants (e.g., `MORTGAGE`, `mortgage`, `Mortgage` → `mortgage`)

This standardization was applied to:

- `repayment_term`  
- `employment_tenure`  
- `housing_status`  
- `income_verification_status`  
- `origination_month`  
- `outcome_status`  
- `loan_purpose`  
- `region_code`

---

### 4. Consolidate and correct specific categorical fields

#### **Housing status**
The diagnostics showed duplicated categories due to inconsistent capitalization.  
All values were standardized to:

- `mortgage`  
- `rent`  
- `own`  
- `any` (rare category)

#### **Income verification status**
Two categories represented the same meaning:

- `source verified`  
- `verified`

These were merged into a single category:

- **`verified`**

#### **Loan purpose**
Duplicate categories existed due to inconsistent casing (e.g., `debt_consolidation` appearing twice).  
All values were standardized to lowercase and deduplicated.

#### **Region code**
State codes appeared in both uppercase and lowercase.  
All values were converted to lowercase for consistency.

---

### 5. Convert `origination_month` to datetime

`origination_month` was stored as a text label (e.g., `"Jul-2015"`).  
To enable time‑based feature engineering, it was converted to a proper datetime format:

- `"Jul-2015"` → `2015-07-01`  
- `"Oct-2015"` → `2015-10-01`  
- etc.

Invalid or unparseable values were set to `NaT`.

---

### 6. Clean and convert `debt_to_income_ratio` to numeric

Diagnostics revealed that this variable contained:

- numeric strings (`"18.0"`, `"24.0"`)  
- corrupted values (`"bad"`)  
- missing values (`NaN`)

To correct this:

- `"bad"` was replaced with `NaN`  
- the column was converted to numeric using `pd.to_numeric(errors='coerce')`

This ensures the variable is ready for imputation and modeling.

---

### 7. Final categorical dataset status

After applying all cleaning steps:

- All categorical variables are standardized, lowercase, and stripped of whitespace  
- Redundant variables have been removed  
- Datatypes are consistent across the dataset  
- `origination_month` is now a datetime variable  
- `debt_to_income_ratio` is now numeric  

In [27]:
# -----------------------------
# 1. Remove _text suffix for categorical variables
# -----------------------------

apps.rename(columns={
    'repayment_term_text': 'repayment_term',
    'employment_tenure_text': 'employment_tenure'
}, inplace=True)

# -----------------------------
# 2. Drop redundant categorical variable
# risk_grade_band gives same info as risk_grade_detail
# -----------------------------

apps.drop(columns=['risk_grade_band'], inplace=True)

# -----------------------------
# 3. Standardize categorical variables (strip + lowercase)
# -----------------------------

cat_cols_to_clean = [
    'repayment_term',
    'employment_tenure',
    'housing_status',
    'income_verification_status',
    'origination_month',
    'outcome_status',
    'loan_purpose',
    'region_code'
]

for col in cat_cols_to_clean:
    apps[col] = apps[col].astype('string').str.strip().str.lower()

# -----------------------------
# 4. Fix housing_status duplicates (mortgage vs MORTGAGE, etc.)
# -----------------------------

apps['housing_status'] = apps['housing_status'].replace({
    'mortgage': 'mortgage',
    'rent': 'rent',
    'own': 'own'
})

# -----------------------------
# 5. Fix income_verification_status
# merge "source verified" + "verified" → "verified"
# -----------------------------

apps['income_verification_status'] = apps['income_verification_status'].replace({
    'source verified': 'verified',
    'verified': 'verified'
})

# -----------------------------
# 6. Convert origination_month to datetime
# -----------------------------

apps['origination_month'] = pd.to_datetime(apps['origination_month'], format='%b-%Y', errors='coerce')

# -----------------------------
# 7. Clean debt_to_income_ratio
# replace "bad" with NaN and convert to numeric
# -----------------------------

apps['debt_to_income_ratio'] = apps['debt_to_income_ratio'].replace('bad', pd.NA)
apps['debt_to_income_ratio'] = pd.to_numeric(apps['debt_to_income_ratio'], errors='coerce')


In [29]:
apps.info()

<class 'pandas.DataFrame'>
RangeIndex: 85000 entries, 0 to 84999
Data columns (total 20 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   requested_amount_usd        85000 non-null  float64       
 1   repayment_term              85000 non-null  string        
 2   monthly_installment_usd     85000 non-null  float64       
 3   last_payment_amount_usd     85000 non-null  float64       
 4   risk_grade_detail           85000 non-null  str           
 5   employment_tenure           76324 non-null  string        
 6   housing_status              85000 non-null  string        
 7   annual_income_usd           84320 non-null  float64       
 8   income_verification_status  85000 non-null  string        
 9   origination_month           85000 non-null  datetime64[us]
 10  outcome_status              85000 non-null  string        
 11  loan_purpose                85000 non-null  string        
 12  r

In [30]:
apps.select_dtypes(include=['object', 'string']).describe()

,repayment_term,risk_grade_detail,employment_tenure,housing_status,income_verification_status,outcome_status,loan_purpose,region_code
count,85000,85000,76324,85000,85000,85000,85000,85000
unique,2,35,11,4,2,2,12,50
top,36 months,C4,10+ years,mortgage,verified,fully paid,debt_consolidation,ca
freq,59223,5289,26275,39118,63539,42500,50891,11842


In [31]:
apps.select_dtypes(include=['int64', 'float64']).describe()

,requested_amount_usd,monthly_installment_usd,last_payment_amount_usd,annual_income_usd,debt_to_income_ratio,credit_score_low,credit_score_high,open_credit_lines,public_records_count,interest_rate,revolving_utilization
count,85000.000000,85000.000000,85000.000000,8.432000e+04,83955.000000,85000.000000,85000.000000,85000.000000,85000.000000,85000.000000,84108.000000
mean,14950.852647,446.071413,3604.090989,7.434189e+04,19.682500,692.034412,696.034482,11.920871,0.248718,13.600630,52.342891
std,8735.147461,258.523612,6222.901104,7.732271e+04,13.170892,29.787475,29.787839,5.670095,0.656437,4.886372,24.293508
min,1000.000000,30.120000,0.000000,0.000000e+00,0.000000,660.000000,664.000000,1.000000,0.000000,5.310000,0.000000
25%,8000.000000,259.955000,308.220000,4.500000e+04,12.800000,670.000000,674.000000,8.000000,0.000000,9.990000,34.200000
50%,13050.000000,382.870000,598.895000,6.300000e+04,18.990000,685.000000,689.000000,11.000000,0.000000,13.180000,52.800000
75%,20000.000000,587.690000,4130.377500,9.000000e+04,25.910000,705.000000,709.000000,15.000000,0.000000,16.550000,70.900000
max,40000.000000,1587.230000,40551.270000,8.900060e+06,999.000000,845.000000,850.000000,70.000000,49.000000,30.990000,136.400000
